web/src/message_events.ts

```diff
@@ -160,8 +167,12 @@ export let update_views_filtered_on_message_property = (
                     continue;
                 }
 
-                const first_id = msg_list.first().id;
-                const last_id = msg_list.last().id;
+                const first_message = msg_list.first();
+                assert(first_message !== undefined);
+                const first_id = first_message.id;
+                const last_message = msg_list.last();
+                assert(last_message !== undefined);
```

Comment:
Do we know these aren’t `undefined`?

Reply:
If they had been, the previously existing code would have given an error for calling `.id` on `undefined`

Reply:
Yeah I'm not sure the previous code is correct here; @amanagr can investigate as a follow-up.